### ISSUE TITLE
Scan with camera for starting position feature

### ISSUE THREAD

**Issue by alexisdurieux** [lichess-org/mobile#410]
Hello,
I would like to be able to scan a position from a book to start study it on a lichess study or to practice against the computer. I think this could be one good addition to the set of features of lichess and I am willing to develop if you are willing for this feature to be implemented.

Thanks

**Comment by veloce** [comment]
Hi! 

Sure this is a nice feature. I'd have to consider the maintenance cost as for every feature but I'd like to see that in the app.

Have you an idea of libraries, ML models you plan to use?

**Comment by alexisdurieux** [comment]
So, I have looked very quickly and I have found https://github.com/linrock/chessboard-recognizer. 
However, I need to investigate further as we probably want the simplest thing. I have to try things to see if that works with Tensorflow lite. Or maybe we d prefer in any case have it connected to the backend so that it benefits the desktop version :)
I ll try to come back in the upcoming days with a more concrete plan.

**Comment by veloce** [comment]
Ideally it should be able to recognise offline, so without a backend connection.

**Comment by ijm8710** [comment]
Would this be like chessvision.ai? If so I would love this as chessvision seems to slowly moving to a premium model.

I wouldn't even be opposed to the idea of asking if Pawel (creator of chessvision) is interested in partnering up

**Comment by nurse-the-code** [comment]
Yeah, honestly, it would be better to have an external open source library that handles the AI heavy-lifting. Then lichess could just call the library. This would allow it to be used in other apps as well. The app should take both screenshots and photos.
